### PR TITLE
Add meeting by clicking on the calendar matrix

### PR DESCRIPTION
**Pull request by pypingou**

### CODE REVIEW

File: fedocal/__init__.py [hunk 1]
```diff
@@ -864,9 +870,30 @@ def add_meeting(calendar_name):
...
+        if mtg_date:
+            form.meeting_date.data = parser.parse(mtg_date).date()
+            form.meeting_date_end.data = parser.parse(mtg_date).date()
+
+        mtg_time = flask.request.args.get('time', None)
+        if mtg_time:
+            form.meeting_time_start.data = mtg_time
+            hours, minutes = mtg_time.split(':')[:2]
+            try:
+                hours = int(hours) + 1
```
Comment (ralphbean) [inline fedocal/__init__.py]: Why the ``+ 1`` here?

File: fedocal/__init__.py [hunk 2]
```diff
@@ -864,9 +870,30 @@ def add_meeting(calendar_name):
...
+            except:
+                pass
```
Comment (ralphbean) [inline fedocal/__init__.py]: What exceptions do you anticipate here?

File: fedocal/__init__.py (hunk 1 above)
Comment (pypingou) [inline fedocal/__init__.py]: So that the end time of the meeting is 1 hour after the start time by default

File: fedocal/__init__.py (hunk 2 above)
Comment (pypingou) [inline fedocal/__init__.py]: If someone puts ``?time=aa:00`` in the URL :)

Comment (ralphbean) [inline fedocal/__init__.py]: Heh.  Then about ``?time=01:aa``?

And, in this case, you let the ``"aa"`` pass through to the code below.  Should it instead be rejected or overwritten?

File: fedocal/__init__.py (hunk 1 above)
Comment (ralphbean) [inline fedocal/__init__.py]: Hm, what if a user calls this with a meeting time that they really did intend.  Then you bump their meeting time up by an hour.

Perhaps this default suggestion should be set in the javascript code that calls this endpoint.

File: fedocal/__init__.py (hunk 2 above)
Comment (pypingou) [inline fedocal/__init__.py]: Good point, I guess d8b0af6 should take care of it

File: fedocal/__init__.py (hunk 1 above)
Comment (pypingou) [inline fedocal/__init__.py]: So the user click on the agenda matrix, the time is retrieved by JS, given to the URL called, we parse it here to do a small data checking, then it is put in the form.
In the add_meeting template we extract the value from the form (if there is one) and insert it into the timespinner().

This way, if you click on a meeting at 18:00, the default time you get are 18:00 start - 19:00 end, but this does not prevent you in anyway from adjusting the times as you would currently.

Comment (pypingou) [inline fedocal/__init__.py]: For the template logic: https://github.com/fedora-infra/fedocal/pull/168/files#diff-38faee44bdff0a8e66fb3ddb82715945R91

Comment (ralphbean) [inline fedocal/__init__.py]: OK, I see now.  :)

**Comment by ralphbean** [comment]
:+1:

**Comment by pypingou** [comment]
Thanks @ralphbean !

**Pull request merged**